src/dimension.c

```diff
@@ -995,8 +1001,32 @@ dimension_add_from_info(DimensionInfo *info)
 
 	Assert(info->ht != NULL);
 
-	dimension_insert(info->ht->fd.id, info->colname, info->coltype,
-					 info->num_slices, info->partitioning_func, info->interval);
+	info->dimension_id = dimension_insert(info->ht->fd.id, info->colname, info->coltype,
+										  info->num_slices, info->partitioning_func, info->interval);
+}
+
+static Datum
+add_dimension_datum(FunctionCallInfo fcinfo, DimensionInfo *info)
```

Comment:
Could we figure out a better name here? It reads as "add datum to dimension" while it is meant to be a datum for `add_dimension`.  Since this is "creating" a datum I think `create_dimension_datum()` would suffice since the this is a datum representing a dimension.

Alternatively, `dimension_create_datum()` given our namespacing/prefixing rules, although this is not a public function so not as important.